File: na-stok-server/src/models/slope.model.ts
import mongoose from "mongoose";

const slopeSchema = new mongoose.Schema({
    address: {
        type: String,
    },
    city: {
        required: true,
        type: String,
    },
    condition_equal: {
        min: 0,
        type: Number,
    },
    condition_max: {
        min: 0,
        type: Number,
    },
    condition_min: {
        min: 0,
        type: Number,
    },
    name: {
        required: true,
        type: String,
      },
    snow_fall: {
        min: 0,
        required: true,
        type: Number,
    },
    status: {
        enum: ["czynny", "nieczynny"],
        required: true,
        type: String,
    },
    update_time: {
        type: Date,
    },
    weather: [{
        clouds: {
            type: String,
        },
        day: {
            required: true,
            type: String,
            validate: {
                validator: (v: string)  => {
                    return /\d(\d|).\d(\d|)/.test(v);
                },
            },
        },
        day_name: {
            required: true,
            type: String,
        },
        temperature: {
            type: Number,
        },
    }],
});

const SlopeModel = mongoose.model("Slopes", slopeSchema);
mongoose.set("useFindAndModify", false);

export default SlopeModel;

File: na-stok-flutter/na_stok_flutter/lib/models/slope_model.dart
import 'dart:convert';

class Weather {
  final String clouds;
  final String day;
  final String dayName;
  final int temperature;

  Weather(this.clouds, this.day, this.dayName, this.temperature);

  Weather.fromJson(Map<String, dynamic> json)
    : clouds = json['clouds'],
      day = json['day'],
      dayName = json['day_name'],
      temperature = json['temperature'];

  @override
  String toString() => '{clouds: $clouds, day: $day, dayName: $dayName, temperature: $temperature}';

}

class Slope {
  final String id;
  final String address;
  final String city;
  final int conditionEqual;
  final int conditionMax;
  final int conditionMin;
  final String name;
  final int snowFall;
  final String status;
  final String updateTime;
  final List<Weather> weatherList;

  Slope.fromJson(Map<String, dynamic> json)
      : name = json['name'],
        id = json['_id'],
        city = json['city'],
        address = json['address'],
        conditionEqual = json['condition_equal'],
        conditionMin = json['condition_min'],
        conditionMax = json['condition_max'],
        updateTime = json['update_time'],
        status = json['status'],
        snowFall = json['snow_fall'],
        weatherList = (json['weather'] as List).map( (e) => Weather.fromJson(e)).toList();

  @override
  String toString() => '{ _id: $id, name: $name, address: $address, city: $city, conditionMin: $conditionMin, conditionMax: $conditionMax, conditionEqual: $conditionEqual,'
      'snowFall: $snowFall, status: $status, updateTime: $updateTime, weatherList: $weatherList}';
}